activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -305,7 +305,7 @@ def column(name, type, options = {})
         type = type.to_sym if type
         options = options.dup
 
-        if @columns_hash[name] && @columns_hash[name].primary_key?
+        if @columns_hash[name] && @columns_hash[name][:primary_key]
```

Comment:
This usage feels like a regression. It's nice to call a clear predicate method to ask about the column's attributes rather than digging in to the column options to fetch the data.